lib/puppet/pops/adapters.rb

```diff
@@ -91,6 +91,7 @@ def extract_tree_text
       last = first + @adapted.length
       @adapted.eAllContents.each do |m|
         m_offset = m.offset
+        next if m_offset.nil?
```

Comment:
Not a good sign that some expressions lack offset. If it is last in a sequence, part of the source will be missing. Think we need to figure out what causes it to be nil. and see if we can set the offset and length. OTOH, it may be only special nodes where the structure is known to continue and skipping will always be correct.